conan/internal/api/install/generators.py

```diff
@@ -151,6 +152,14 @@ def _receive_conf(conanfile):
             conanfile.conf.compose_conf(build_require.conf_info)
 
 
+def _receive_generators(conanfile):
+    """  Collect generators_info from the immediate build_requires"""
+    for build_require in conanfile.dependencies.direct_build.values():
+        if build_require.generator_info:
+            # In case generators has been defined as a tuple
+            conanfile.generators = list(conanfile.generators) + list(build_require.generator_info)
```

Comment:
Generators from dependencies should be executed before the ones from the current conanfile, to let space for the consumer to adapt, modify, consume, overwrite the generated files.